editor/project_manager.cpp

```diff
@@ -1400,6 +1401,40 @@ void ProjectManager::_load_recent_projects() {
 		fpath->add_color_override("font_color", font_color);
 		fpath->set_clip_text(true);
 
+		// indicate project compatibility via colored label
+		Label *engine_version = memnew(Label());
+		engine_version->set_name("engine_version");
+		engine_version->set_h_size_flags(SIZE_EXPAND_FILL);
+		Color engine_color = Color(1, 0, 0, 0.9);
+		bool indicate_incompat = false;
+
+		// check if project is still in place
+		if (FileAccess::exists(item.conf)) {
+			if (item.config_version < ProjectSettings::CONFIG_VERSION) {
+				indicate_incompat = true;
+				engine_color = Color(1, 1, 0.2, 0.5);
+				engine_version->set_text(TTR("Older version project"));
+			}
+			if (item.config_version > ProjectSettings::CONFIG_VERSION) {
+				indicate_incompat = true;
+				engine_version->set_text(TTR("Incompatible"));
+			}
+		} else {
+			indicate_incompat = true;
+			engine_version->set_text(TTR("Deleted"));
+		}
+
+		// show warning if something is odd
+		if (indicate_incompat) {
```

Comment:
It's quite wasteful to allocate `engine_version` for every project, and discard it if not needed. Apart from short transitions periods between releases where the `config_version` changed (which should not happen often, it's not linked to the engine version), most users won't have incompatible projects.

It would be cleaner and more efficient to check first if a label is needed, and instantiate it only then.